Use timestamp as path prefix for aggregated js file

nabcos opened the pull request:
This is WIP and not working correctly right now, more a reminder for future work and RFC whether someone spots what's wrong here. The reason for this is better cache busting in browsers.

For some reason the ExternalStaticResources is not attached to restlet correctly. It's processed in [NexusApplication](https://github.com/sonatype/nexus/blob/timestamped-resource-url/nexus/plugins/restlet1x/nexus-restlet1x-plugin/src/main/java/org/sonatype/nexus/rest/NexusApplication.java#L210), but still gives 404 for the timestamped URLs.

This is a hack anyway because it uses an ExternalStaticResource for sonatype-all.js. It should rather use classpath lookup, but that seems to not work at all for the location of the js files here (they are included in the WAR root directory). This will get better when we can wrap the UI stuff into a plugin.

nabcos committed:
Use timestamp as path prefix for aggregated js file

jdillon commented:
Why not move the core js muck this needs to load in to the WEB-INF/classes directory?

nabcos commented:
... Because it's too obvious? I'll try that tomorrow.

nabcos committed:
Use classpath lookup for main js/css files

nabcos committed:
Cache content of resource files

Just a test change to see whether Nexus returns 404 because it cannot
read from the resource later on. Initial debugging shows content is
cached without problems.

nabcos commented:
Changes squashed into ui-extjs3 branch.
